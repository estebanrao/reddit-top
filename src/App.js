import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectConfigIsDarkModeActive } from './redux/config/config.selectors';

import {
  unstable_createMuiStrictModeTheme as createMuiTheme,
  ThemeProvider,
} from '@material-ui/core/styles';

import TopBar from './components/top-bar/top-bar.component';
import PostsList from './components/posts-list/posts-list.component';
import PostDetails from './components/post-details/post-details.component';

import CssBaseline from '@material-ui/core/CssBaseline';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
  },
  drawerPaper: {
    width: '100vw',
    [theme.breakpoints.up('md')]: {
      width: '50vw',
    },
  },
}));

export const App = ({ isDarkModeActive }) => {
  const classes = useStyles();

  const palletType = isDarkModeActive ? 'dark' : 'light';
  const darkTheme = createMuiTheme({
    palette: {
      type: palletType,
    },
  });

  // console.log('Render: App');
  return (
    <ThemeProvider theme={darkTheme}>
      <div className={classes.root}>
        <CssBaseline />
        <TopBar />
        <PostsList />
        <PostDetails />
      </div>
    </ThemeProvider>
  );
};

const mapStateToProps = createStructuredSelector({
  isDarkModeActive: selectConfigIsDarkModeActive,
});

export default connect(mapStateToProps)(App);
