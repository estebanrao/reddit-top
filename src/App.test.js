import { shallow } from 'enzyme';
import React from 'react';
import { App } from './App';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<App />);
});

it('should render App component', () => {
  expect(wrapper.debug()).toMatchSnapshot();
});
