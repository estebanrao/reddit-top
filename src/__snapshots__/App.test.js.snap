// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component 1`] = `
"<ThemeProvider theme={{...}}>
  <div className=\\"makeStyles-root-1\\">
    <WithStyles(CssBaseline) />
    <Connect(TopBar) />
    <Connect(PostsList) />
    <Connect(PostDetails) />
  </div>
</ThemeProvider>"
`;
